JASP-Engine/JASP/R/commonbayesianttest.R

```diff
@@ -0,0 +1,1786 @@
+		  }
+
+		} else {
+
+		  if (is.null(options[["pairs"]])) {
+		    for (var in dependents) {
+		      errors[[var]] <- .hasErrors(dataset, perform = "run", message = 'short',
+		                                  type = c('infinity','observations','variance'),
+		                                  all.target = var, observations.amount = "< 2")
+		    }
+		  } else {
+
+		    for (var in dependents) {
+		      pair <- options[["pairs"]][[var]]
+
+		      if (pair[[1]] == "" || pair[[2]] == "") {
+
+		        errors[[var]] <- list(message = sprintf("Please provide another variable."))
+
+		      } else if (identical(pair[[1]], pair[[2]])) {
+
+		        errors[[var]] <- list(message = sprintf("Variables %s and %s are the same!",
+		                                                pair[[1]], pair[[2]]))
+
+		      } else {
+
+		        errors[[var]] <- .hasErrors(dataset, perform = "run", message = 'short',
+		                                    type = c('infinity','observations','variance'),
+		                                    all.target = c(pair[[1]],pair[[2]]), observations.amount = "< 2")
+		      }
+		    }
+		  }
+		}
+
+		return(errors)
+	} else {
+		return(NULL)
+	}
+}
+
+.ttestBayesianInitBayesFactorPackageOptions <- function() {
+
+  # sets all options required for the BayesFactor package.
+  # note that options$... / options[[...]] and options(...) are two COMPLETELY DIFFERENT things!!
+  # also, options(...) is, by R's default, global so this works
+
+  if(is.null(options("BFMaxModels")))         options(BFMaxModels = 50000)
+  if(is.null(options("BFpretestIterations"))) options(BFpretestIterations = 100)
+  if(is.null(options("BFapproxOptimizer")))   options(BFapproxOptimizer = "optim")
+  if(is.null(options("BFapproxLimits")))      options(BFapproxLimits = c(-15,15))
+  if(is.null(options("BFprogress")))          options(BFprogress = interactive())
+  if(is.null(options("BFfactorsMax")))        options(BFfactorsMax = 5)
+
+}
+
+.ttestBayesianInitOptions <- function(jaspResults, options, analysis = c("independent", "paired", "one-sample")) {
+
+  # initialize options: takes user input and determines:
+  #
+  # - can any analysis be done?
+  # - which variables will need to be computed for plots?
+  #
+  # in addition:
+  #
+  # - unifies the hypothesis type (larger, smaller or equal) inside options[["oneSided"]]
+  # - defines the dependencies for all objects and puts this inside options[["stateKey"]]
+
+
+  analysis <- match.arg(analysis)
+  options[["ttestType"]] <- analysis
+
+  # dependencies specified here to pass as depends
+	defaults <- c(
+	  "effectSizeStandardized", "groupingVariable", "hypothesis",
+	  "informativeCauchyLocation", "informativeCauchyScale", "informativeNormalMean",
+	  "informativeNormalStd", "informativeStandardizedEffectSize",
+	  "informativeTDf", "informativeTLocation", "informativeTScale",
+	  "missingValues", "priorWidth", "testStatistic", "wilcoxonSamplesNumber"
+	)
+  options[["stateKey"]] <- list(
```

Comment:
I think this function overcomplicates things a little. It makes the state dependencies rely on a lot of little steps  and you have to switch files to find out what the actual table depends on.

So instead of the following (which I think contains a duplicate "pairs"):
```
(line 135 in commonbayesianttest.R)
defaults <- c(
  "effectSizeStandardized", "groupingVariable", "hypothesis",
  "informativeCauchyLocation", "informativeCauchyScale", "informativeNormalMean",
  "informativeNormalStd", "informativeStandardizedEffectSize",
  "informativeTDf", "informativeTLocation", "informativeTScale",
  "missingValues", "priorWidth", "testStatistic", "wilcoxonSamplesNumber"
)

(line 142 in commonbayesianttest.R)
options[["stateKey"]] <- list(ttestResults = defaults)

(line 202 in commonbayesianttest.R)
options[["stateKey"]][["ttestResults"]] <- c(options[["stateKey"]][["ttestResults"]], "pairs")

(line 49 in ttestbayesianpairedsamples.R)
dependencies <- options[["stateKey"]][["ttestResults"]]
ttestTable$dependOnOptions(c(dependencies, "bayesFactorType", "pairs"))
```
I would propose just doing this:
```
(line 49 in ttestbayesianpairedsamples.R)
ttestTable$dependOnOptions("pairs", "bayesFactorType", "effectSizeStandardized", "groupingVariable", "hypothesis",
  "informativeCauchyLocation", "informativeCauchyScale", "informativeNormalMean",
  "informativeNormalStd", "informativeStandardizedEffectSize",
  "informativeTDf", "informativeTLocation", "informativeTScale",
  "missingValues", "priorWidth", "testStatistic", "wilcoxonSamplesNumber")
```

Reply:
I guess this means these options are used in all analyses using this common functionality? Because if an option is specified but it doesn't exist in an analysis then it will always regard it as "rerun".